plugins/modules/device_credential_workflow_manager.py

```diff
@@ -2087,214 +2100,232 @@ def get_want_assign_credentials(self, AssignCredentials):
-                            httpReadDetail = item
-                    if not httpReadDetail:
-                        self.msg = "The description and username for the httpRead credential are invalid."
+                elif http_read_description and http_read_username:
+                    for item in http_read_details:
+                        if item.get("description") == http_read_description and \
+                                item.get("username") == http_read_username:
+                            http_read_detail = item
+
+                    if not http_read_detail:
+                        self.msg = "The description and username for the http_read credential are invalid."
                         self.status = "failed"
                         return self
+
                 if current_ccc_version_as_int <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
-                    want.get("assign_credentials").update({"httpRead": httpReadDetail.get("id")})
+                    want.get("assign_credentials").update({"httpRead": http_read_detail.get("id")})
                 else:
                     want.get("assign_credentials").update({
-                        "httpReadCredentialsId": {"credentialsId": httpReadDetail.get("id")}
+                        "httpReadCredentialsId": {"credentialsId": http_read_detail.get("id")}
                     })
 
-        https_write = AssignCredentials.get("https_write")
+        https_write = assign_credentials.get("https_write")
         if https_write:
-            httpWriteId = https_write.get("id")
-            httpWriteDescription = https_write.get("description")
-            httpWriteUsername = https_write.get("username")
-            if httpWriteId or httpWriteDescription and httpWriteUsername:
+            http_write_id = https_write.get("id")
+            http_write_description = https_write.get("description")
+            http_write_username = https_write.get("username")
 
+            if http_write_id or http_write_description and http_write_username:
                 # All httpWrite details from the Cisco Catalyst Center
-                httpWrite_details = global_credentials.get("httpsWrite")
-                if not httpWrite_details:
-                    self.msg = "Global httpWrite credential is not available."
+                http_write_details = global_credentials.get("httpsWrite")
+                if not http_write_details:
+                    self.msg = "Global http_write credential is not available."
                     self.status = "failed"
                     return self
-                httpWriteDetail = None
-                if httpWriteId:
-                    httpWriteDetail = get_dict_result(httpWrite_details, "id", httpWriteId)
-                    if not httpWriteDetail:
-                        self.msg = "The ID of the httpWrite credential is not valid."
+                http_write_detail = None
```

Comment:
Same... we are overwriting the value that we have already got in line 2263....

Reply:
Here also, it's  http_write_detail and  http_write_details 
now changed to global_https_write_details